packages/agreement-process/src/services/agreementStampUtils.ts

```diff
@@ -29,9 +34,17 @@ export const suspendedByProducerStamp = (
   agreement: Agreement,
   requesterOrgId: TenantId,
   destinationState: AgreementState,
-  stamp: AgreementStamp
+  stamp: AgreementStamp,
+  delegateProducerId?: TenantId | undefined
 ): AgreementStamp | undefined =>
-  match([requesterOrgId, destinationState])
-    .with([agreement.producerId, agreementState.suspended], () => stamp)
+  match<[TenantId | undefined, AgreementState]>([
+    requesterOrgId,
+    destinationState,
+  ])
+    .with(
+      [agreement.producerId, agreementState.suspended],
+      [delegateProducerId, agreementState.suspended],
+      () => stamp
+    )
     .with([agreement.producerId, P.any], () => undefined)
```

Comment:
Should we do the same in this second match?

```suggestion
    .with(
      [agreement.producerId, P.any],
      [delegateProducerId, P.any],
      () => undefined
    )
```